pkg/provider/azure_loadbalancer.go

```diff
@@ -693,19 +714,25 @@ func (az *Cloud) selectLoadBalancer(clusterName string, service *v1.Service, exi
 	return selectedLB, existsLb, nil
 }
 
-func (az *Cloud) getServiceLoadBalancerStatus(service *v1.Service, lb *network.LoadBalancer) (status *v1.LoadBalancerStatus, fipConfig *network.FrontendIPConfiguration, err error) {
+// getServiceLoadBalancerStatus returns LB status for the Service.
+// Before DualStack support, old logic takes the first ingress IP as non-additional one
+// and the second one as additional one. With DualStack support, the second IP may be
+// the IP of another IP family so the new logic returns two variables.
+func (az *Cloud) getServiceLoadBalancerStatus(service *v1.Service, lb *network.LoadBalancer) (status *v1.LoadBalancerStatus, lbIPsNoAdditionalPIPs []string, fipConfigs []*network.FrontendIPConfiguration, err error) {
```

Comment:
is noAdditional means primary?

Reply:
It means compared to ingress IPs in `status`, `lbIPsNoAdditionalPIPs` doesn't include those IPs from `service.beta.kubernetes.io/azure-additional-public-ips` annotation.

Reply:
Will it be better to replace to primary?

Reply:
Yes, `NoAdditional` seems too straightforward. Replaced.